# Profitable App Profile Analysis for the App Store and Google Play


In this project we will analyse what makes an mobile app popular and how this could be in used in order to built more profitable apps. Our main goal is to come up with an app sugestion that cna be profitable fot both App Store and Google Play markets. The app should be free and make its profit out of advertising and because of that the we need to analyse the data to see what makes an app popular.

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

Collecting data for over four million apps requires a significant amount of time and money, so we'll try to analyze a sample of data instead. To avoid spending resources with collecting new data ourselves, we should first try to see whether we can find any relevant existing data at no cost. Luckily, these are two data sets that seem suitable for our purpose:

A data set containing data about approximately ten thousand Android apps from Google Play. You can download the data set directly from [this link](https://www.kaggle.com/lava18/google-play-store-apps).

A data set containing data about approximately seven thousand iOS apps from the App Store. You can download the data set directly from [this link](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps).

**Disclaimer: This is guided project from the DataQuest's 'Python for Data Science: Fundamentals' course developed by learning purposes. Although it may look like other projects made for the same reason, this project has some features of its own implemented by me. If you are going to read this project all the way, keep in mind that this is my first project of this kind.**

## Opening the data

We wil begin by opening both datasets usign the `open ()` and `reader()` 

In [1]:
from csv import reader
#Opening the Apple Store data
opened_file1 = open('AppleStore.csv', encoding='utf8')
read_file1 = reader(opened_file1)
ios = list(read_file1)
ios_header = ios[0]
ios_data = ios[1:]

#Openning the Play Store data
opened_file2 = open('googleplaystore.csv', encoding='utf8')
read_file2 = reader(opened_file2)
android = list(read_file2)
android_header = android[0]
android_data = android[1:]


Now we're going to write a function to make the exploring of the data easier. A option to show the total number of rows and colunms is also included.

In [29]:
def explore_data(data_set, start, end, rows_colunms=False):
    sliced_dataset = data_set[start:end] #Slices the dataset
    for row in sliced_dataset:
        print(row)
        print('\n') 
    if rows_colunms:
        print('Total number of colunms: ', len(data_set[0]))
        print('Total number of rows: ', len(data_set))
        
print(ios_header)
print('\n')
explore_data(ios_data, 0, 6, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


['5', '282935706', 'Bible', '92774400', 'USD', '0', '985920', '5320', '4.5', '5', '7.5.1', '4+', 'Reference', '37'

In [3]:
print(android_header)
print('\n')
explore_data(android_data,0 , 6, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

# Deleting wrong data

It has been reported in the [Google Play data set dedicated discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) that one of the rows (row 10472) is missing the 'Category' colunm, what makes all the other colunms to shift and because of that this row has less colunms than the it should have. We will print that row next with the android header and android first row, which is correct.

In [4]:
print(android_header) # Header
print('\n')
print(android_data[10472]) # Incorrect row
print('\n')
print(android_data[0]) # Correct row

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


In order to verify if this mistake happens in any other row, we'll write a function that will compare the length of every row in each dataset to the length of the data set's header.

In [5]:
def checking_rows(dataset):
    count = 0
    for row in dataset:
        if len(row) != len(dataset[0]):
            print(row)
                
                
checking_rows(android)
checking_rows(ios)

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


Now we can be sure that row 10472 of the Google Store database is the only wrong row. We'll now delete it.

In [6]:
print(len(android_data))
del android_data[10472] # This line cannot be executed twice
print(len(android_data))

10841
10840


# Duplicates


We'll now check both databases in order to find if they present duplicate data. This task willbe divided in two steps. First, we are going to check if there is duplicate data and then we'll delete it.


### First Step

We will wirte a function tha will use a `for` loop to check every row in the dataset and assign them a to list of unique app or to a list of duplicated apps.


In [7]:
def duplicates(dataset):
    unique_data = []
    duplicated_data = []
    for row in dataset:
        name = row[0]
        if name in unique_data:
            duplicated_data.append(name)
        else:
            unique_data.append(name)
    return duplicated_data

duplicated_android = duplicates(android_data)
duplicated_ios = duplicates(ios_data)
print('Number of android duplicated data: ', len(duplicated_android))
print('Number of iOS duplicated data: ', len(duplicated_ios))

Number of android duplicated data:  1181
Number of iOS duplicated data:  0


There are 1181 duplicated apps in the android database and none in the IOS database.

In [8]:
print('Example of duplicated data: \n\n', duplicated_android[:16])

Example of duplicated data: 

 ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software', 'MailChimp - Email, Marketing Automation']


### Second Step

Using the Google Ads app as an example, we can see that the only difference between the duplicated data is in the fourth colunm, which correspond to the number of reviews. In order to not delete th eduplicated data randomly, we'll use this criteria to do it. For each duplicated app, we'll only keep only the row with the highest number of reviews because the higher the number, the most recent and reliable the ratings.

In [9]:
for row in android_data:
    name = row[0]
    if name == 'Google Ads':
        print(row)

['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29331', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


In order to keep only the row with the highest numer of reviews for each duplicated data, we'll create a dictionary where each key is a unique app name, and the value is the highest number of reviews fo that app.

In [10]:
reviews_max = {}

for row in android_data:
    name = row[0]
    n_reviews = float(row[3])
    
    if name in reviews_max:
        if n_reviews > reviews_max[name]:
            reviews_max[name] = n_reviews
    else:
        reviews_max[name] = n_reviews

Now let's check if the dictionary length matches the total number of unique apps.

In [11]:
print('Number of unique apps: ', len(android_data) - len(duplicated_android))
print('Dictionary length: ', len(reviews_max))

Number of unique apps:  9659
Dictionary length:  9659


We'll now create a new android database with only the unique apps. For that we will initialize two empty lists, `android_clean` and `already_added`. We'll then loop through the android data set and, for each iteration, add the current row to the `android_clean` list and the app name to the `already_added` list if:
   * The number o reviews of the current app matches the number of reviwes stored in the `reviews_max` dictonary:
   * The name of the list is not in the `already_added` list, which would mean that this has already been added to the       `android_clean` list. That's importante because the highest number of reviews may not be unique for each duplicated app.

In [12]:
android_clean = []
already_added = []

for row in android_data:
    name = row[0]
    n_reviews = float(row[3])
    
    if reviews_max[name] == n_reviews and name not in already_added:
        android_clean.append(row)
        already_added.append(name)

Let's quickly explore the new data set and check if its length matches the number of unique apps.

In [13]:
explore_data(android_clean, 0, 5, True)
print('Number of unique apps: ', len(android_data) - len(duplicated_android))

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Total number of colunms:  13
Total number of rows:  9659
Number of unique apps:  9659


Duplicate apps removed successfully.

# Removing Non-Enlgish apps

If you explore the data set long enoguh, you'll notice that the names of some apps suggest they are not directed to the english-speaking audience, as we can see in the examples below:

In [14]:
print(ios_data[814][2])

print(android_clean[4412][0])

搜狐新闻—新闻热点资讯掌上阅读软件
中国語 AQリスニング


### First step

As we're not interested in these apps, we will now write a function identify them. We'll take advantage of the [ASCII standard](https://pt.wikipedia.org/wiki/ASCII) and the built-in `ord()` to do it.

In [15]:
def is_english(a_string):
    for characther in a_string:
        if ord(characther) > 127:
            return False
    return True

print(is_english('Facebook'))
print(is_english('中国語 AQリスニング'))

True
False


The function seems to be working just fine, but it still has some problems as it cannot identify some english app names as we can see in the example below:

In [16]:
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

False
False


That happen because emoji and characters  like `™` are outside the ASCII range we determined. We'll then adapt `is_english` to return `False` only if the string has more than three characters that fall outside the ASCII range (0 - 127). That is necessary so we can minimiza the amount of data loss.

In [17]:
def is_english(a_string):
    count = 0
    for characther in a_string:
        if ord(characther) > 127:
            count += 1
    if count > 3:
        return False
    return True

print(is_english('Facebook'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))
print(is_english('中国語 AQリスニング'))

True
True
True
False


Altough the function is not perfect, it is now good enough to minimize data loss during the cleaning process.

### Second step

We will now take advantage of the `is_english` function to filter the non-english apps from both data sets.

In [18]:
android_en = []
ios_en = []

for row in android_clean:
    name = row[0]
    if is_english(name):
        android_en.append(row)

for row in ios_data:
    name = row[2]
    if is_english(name):
        ios_en.append(row)

explore_data(android_en, 0, 5, True)
print('\n')
explore_data(ios_en, 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Total number of colunms:  13
Total number of rows:  9614


['1', '281656475', 'PAC-MAN Prem

We're now left with 9614 Android apps and 6183 iOS apps.

# Filtering free apps

For this project, we are only interested in analyzing the free apps. Therefore, we'll loop through both data sets and fill the `android_free` and `ios_free` lists with only free apps.

In [19]:
android_free = []
ios_free = []

for row in android_en:
    price = row[7]
    if price == '0':
        android_free.append(row)

for row in ios_en:
    price = float(row[5])
    if price == 0:
        ios_free.append(row)

print('Android free apps: ', len(android_free))
print('iOS free apps: ', len(ios_free))

Android free apps:  8864
iOS free apps:  3222


Finally, we're down to 8864 Android apps and 3222 iOS apps.

## Most popular apps by genre

As mentioned in the introduction, the main goal of this project is dertermine what kind of app is more likely to attrac more users. That is important beacause the revenue of a free app comes basic from advertisement. Therefore, the higher the number of users, the higher the revenue.
Considering that the end goal is to add the app both on Google Play and App Store, it is necessary to find app profiles that are successful in both markets. We'll start by analyzing the apps by genre.

We'll build two functions to generate a frequency table of any colunm of tha data sets. The first one will actually creat the frequency table and second will display it in a descending order. Notice that by default the data will be diplayed in percentage, but it can also be displayed by the actual number of apps.

In [20]:
def freq_table(dataset, index, percentage=True):
    table = {}
    for row in dataset:
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    if not percentage:
        return table
    else:
        percent_table = {}
        for line in table:
            percent = (table[line] / len(dataset)) * 100
            percent_table[line] = percent
        return percent_table
table_ios = freq_table(ios_free, 12, True)
    
def display_table(table):   
    table_display = []
    for key in table:
        key_tuple = (table[key], key)
        table_display.append(key_tuple)
    table_sorted = sorted(table_display, reverse=True)
    for tuple in table_sorted:
        print(tuple[1], ':', tuple[0])

display_table(table_ios)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can easily see the games dominate the English free apps market (58.16%). Entertaiment apps are close to 8% while Photo & Video apps do not reach 5%. Education (3,66%) and Social Networking (3.28%) and the top five. However, it is not yet possible to take any conlusions form this because the fact that there is a large number of apps for a particular genre does not imply that genre has a large number of users. The damand might not be as large as the offer and, also, a large offer creates a more tough market to succeed.

Now, let's take a look in the Google Play data set. For that, we'll analyse two colunms that seem to be related: the 'Category' and 'Genres' colunms.

In [21]:
android_table1 = freq_table(android_free, 1, True) #Category column
display_table(android_table1) 

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In [22]:
android_table2 = freq_table(android_free, 9) # Genres colunm
display_table(android_table2)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The scenario is completely different for the Google Play data set as its market is more equally divided among the categories and genres than iOS market, altough the Family category corresponds to almost 19% of the apps. Games (9.72%), Tools (8.46%), Business (4.59%) and Lifestyle (3.90%) and a diversified top five. The Android database also seems to have more apps designed for practical life purposes than just for gaming.
Considering the only the 'Genres' colunm, it has a lot more categories, which makes it harder to analyse and take conclusions from. That's why from now on we'll only consider the 'Category' colunm for the Android apps.

### Most popular apps by genre in the App Store

Fisrt, we will loop through each genre and for each genre we'll loop through each app to calculate the average numbers of reviews per app for each genre.

In [23]:
genres_ios = freq_table(ios_free, 12)

rating_per_genre = {}

for genre in genres_ios:
    total = 0
    len_genre = 0
 
    for app in ios_free:
        genre_app = app[12]
        if genre_app == genre:
            n_ratings = float(app[6])
            total += n_ratings
            len_genre += 1
            
    avg_n_ratings = total / len_genre
    rating_per_genre[genre] = avg_n_ratings

display_table(rating_per_genre)
            

Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


We can see that navigation apps have the highest number of user reviews in average, followed byreference and social networking apps averaging over 70,000 reviews per app. However we should check how concentrated are these reviews in one or two big apps. For that, we built a small function that displays the number of reviews per app in a determined genre.

In [24]:
def ios_genre_search(s_genre):
    app_ratings_total = {}
    for app in ios_free:
        genre = app[12]
        if genre == s_genre:
            name = app[2]
            n_reviews = float(app[6])
            app_ratings_total[name] = n_reviews
    display_table(app_ratings_total)
    
    
ios_genre_search('Navigation')
print('\n')
ios_genre_search('Reference')
print('\n')
ios_genre_search('Social Networking')


Waze - GPS Navigation, Maps & Real-time Traffic : 345046.0
Google Maps - Navigation & Transit : 154911.0
Geocaching® : 12811.0
CoPilot GPS – Car Navigation & Offline Maps : 3582.0
ImmobilienScout24: Real Estate Search in Germany : 187.0
Railway Route Search : 5.0


Bible : 985920.0
Dictionary.com Dictionary & Thesaurus : 200047.0
Dictionary.com Dictionary & Thesaurus for iPad : 54175.0
Google Translate : 26786.0
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418.0
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588.0
Merriam-Webster Dictionary : 16849.0
Night Sky : 12122.0
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535.0
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693.0
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497.0
Guides for Pokémon GO - Pokemon GO News and Cheats : 826.0
WWDC : 762.0
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraf

As we can see, the navigation and social networking apps are dominated by few famous apps. Navigation for instance has almost half a million reviews just for Waze and Google Maps. That is why we should consider reference apps as a good option here. This genre reviews are much influenced by the bible app, however this shoud not be a problem since the its number of reviews is so much greater than the others that makes reference apps a one-app genre, which show us a potencial room for new apps to succceed.

We could use the Bible as an example to be replicated with a different famous book, adding new features and making the app even more interactive to the user.

## Most popular apps by genre on Google Play

The Google Play database gives us data about the number of intalls for each app. However, we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.) which make it look less precise.


In [25]:
android_freq = freq_table(android_free, 5)
display_table(android_freq)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


We only intend to get an ideia of which genres attracc the most users, therefore we do not need this data to be abslotely precise. We'll then consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

During this process we'll need to convert the number of reviews to float, so you'll notice that while we calculate the average number of reviews per app in each category, we'll als do this convertion.

In [26]:
category_android = freq_table(android_free, 1)

installs_per_category = {}

for category in category_android:
    total = 0
    len_category = 0
 
    for app in android_free:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
            
    avg_n_installs = total / len_category
    installs_per_category[category] = avg_n_installs

display_table(installs_per_category)
            

COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16787331.344927534
GAME : 15588015.603248259
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
TOOLS : 10801391.298666667
NEWS_AND_MAGAZINES : 9549178.467741935
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.9853479853
MAPS_AND_NAVIGATION : 4056941.7741935486
FAMILY : 3695641.8198090694
SPORTS : 3638640.1428571427
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1833495.145631068
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
BEAUTY : 51315

Communication, Video Players and social apps dominate the Google Play market. We know however that this catgories are alread built around some huge apps like Facebook, Whatsapp and YouTube. Considering that and that we suggested the Reference genre as good option for the iOS market, we should take a look at the Book and Reference category.

In [27]:
total_BaR = 0
for app in android_free:

    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])
        total_BaR += 1
        
print('\n')
print('TOTAL: ', total_BaR)

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

We now know that there are 190 apps in this category. Let's chack how many of them have over 100,000,000 installs.

In [28]:
total_BaR_over = 0
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])
        total_BaR_over += 1
print('\n')
print('Total: ', total_BaR_over)

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


Total:  5


It looks like this category has just a few very popular apps and also a lot of the apps are software for processing and reading ebooks, dictionaries and libraries, which seems to show room for new apps to succeed. 
That's why a similiar ideia to that we suggested for the iOS market may have some potential here. Also, it is part of our goal to suggest the same app to be built for both the plataforms.
Selecting a popular book an making an app out of it, with new features and user's interactions might be the way to go here.

# Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We conclued that creating an app out of a popular book could be profitable for both App Store and Google Play markets. As these markets are already full o libraries and e-readers, a fun interactive app about a book that lots of people care about could really draw some attention, especially if we use nice features and new forms of interation with the book's content.
